src/TurboMqtt/Client/IMqttClient.cs

```diff
@@ -332,8 +332,8 @@ public void DoCancel()
     public async Task<IPublishResult> PublishAsync(MqttMessage message,
         CancellationToken cancellationToken = default)
     {
-        // if (_transport.Status != ConnectionStatus.Connected)
-        //     return new PublishingProtocol.PublishFailure("Not connected to broker.");
+        if (_transport.Status != ConnectionStatus.Connected)
```

Comment:
Stop allowing `PublishPacket`s to be written while we're waiting for the connection to restart - the caller of the `IMqttClient` should be checking these anyway.